zuul_alpha/core.py

```diff
@@ -341,13 +405,15 @@ def generate_key_pair(self):
             f.write(b64encode(ciphertext_blob))
 
         with open(self.public_key_file, 'w') as f:
-            f.write(public_key)
+            f.write(public_key_pem)
 
     def encrypt(self, secret_name, secret, public_key=None):
         '''Single secret encryptor
 
         Takes in a single key-value pair secret and encrypts it with the RSA
-        public key.
+        public key, UNLESS, the secret is too large for RSA payload size (214
```

Comment:
@stlava I agree. After doing some more testing and reading, I think it'll be worthwhile always to do RSA-AES hybrid for which I'll create a separate PR shortly. :
1) The marginal compute time isn't significant
2) RSA-2048 is slightly less secure than AES-128 anyways
3) AES-128 is performant, and according to NIST secure until at least 2030
4) Maintainable standard

- NIST Recommendations: https://www.keylength.com/en/4/
- Discussion: https://crypto.stackexchange.com/questions/10685/hybrid-encryption-with-rsa-and-aes-versus-spliting-into-multiple-rsa-messages

```
RSA's key size is not directly related to its security margin. An RSA 2048 key in your example is equated with 112 bit security by NIST and about 104 bit by ECRYPT-II (2012 predictions). So RSA 2048 is significantly less secure and much slower than any form of AES. Note that RSA encryption also requires access to a secure random number generator to generate the padding.

All in all, both RSA with a key size of 2048 bits and AES-128 are pretty secure. There is a lot of reason to expect that other parts of the system are much more vulnerable to attack. This is especially true if communication is performed without integrity/authenticity provided by a signature or other authentication tag. Once your message has integrity and authenticity protection you need to take care of side channel attacks and attacks on physical security, bugs in software (updates) etc. etc. etc.

Please use a standardized form of hybrid encryption and quickly move on to secure other parts of your system.
```

@stlava